src/component/application/ApplicationEdit.tsx

```diff
@@ -0,0 +1,154 @@
+import HeaderTitle from '../common/HeaderTitle';
+import AccessContext from '../../contexts/AccessContext';
+import useApplicationsApi from '../../hooks/api/actions/useApplicationsApi/useApplicationsApi';
+import useApplication from '../../hooks/api/getters/useApplication/useApplication';
+import { useHistory, useParams } from 'react-router-dom';
+import { useLocationSettings } from '../../hooks/useLocationSettings';
+
+const ApplicationEdit = () => {
+    const history = useHistory();
+    const { name } = useParams<{ name: string }>();
+    const { refetchApplication, application } = useApplication(name);
+    const { appName, url, description, icon = 'apps', createdAt } = application;
+    const { hasAccess } = useContext(AccessContext);
+    const { deleteApplication } = useApplicationsApi();
+    const { locationSettings } = useLocationSettings();
+
+    const [loading, setLoading] = useState(true);
+    const [showDialog, setShowDialog] = useState(false);
+
+    useEffect(() => {
+        refetchApplication();
+        setLoading(false);
+        // eslint-disable-next-line
+    }, []);
+
+    const toggleModal = () => {
+        setShowDialog(!showDialog);
+    };
+
+    const formatDate = (v: Date) =>
+        formatDateWithLocale(v, locationSettings.locale);
+
+    const onDeleteApplication = async (evt: Event) => {
+        evt.preventDefault();
+        await deleteApplication(appName);
+        history.push('/applications');
+    };
+
+    const renderModal = () => (
+        <Dialogue
+            open={showDialog}
+            onClose={toggleModal}
+            onClick={onDeleteApplication}
+            title="Are you sure you want to delete this application?"
+        />
+    );
+    const tabData = [
+        {
+            label: 'Application overview',
+            component: <ApplicationView />,
+        },
+        {
+            label: 'Edit application',
+            component: <ApplicationUpdate application={application} />,
+        },
+    ];
+
+    if (loading) {
+        return (
+            <div>
+                <p>Loading...</p>
+                <LinearProgress />
+            </div>
+        );
+    } else if (!application) {
+        return <p>Application ({appName}) not found</p>;
+    }
+    return (
+        <PageContent
+            headerContent={
+                <HeaderTitle
+                    title={
+                        <span
+                            style={{
+                                display: 'flex',
+                                alignItems: 'center',
+                            }}
+                        >
+                            <Avatar style={{ marginRight: '8px' }}>
+                                <Icon>{icon || 'apps'}</Icon>
+                            </Avatar>
+                            {appName}
+                        </span>
+                    }
+                    actions={
+                        <>
+                            <ConditionallyRender
```

Comment:
I think these can be PermissionIconButtons

Reply:
@FredrikOseberg but there's no permission here. the condition depends if the URL is truthy or falsy.